### PR TITLE
Additional node api

### PR DESCRIPTION
**Pull request by HM4704**
- new API to query the peer addresses of a node
- first implementation for sync info

### CODE REVIEW

File: node/apiserver.go [hunk 1]
```diff
@@ -46,6 +48,46 @@ func (p *ProximaNode) GetNodeInfo() *global.NodeInfo {
...
+	latestSlot, latestHealthySlot, synced := p.workflow.LatestBranchSlots()
+	ret := &api.SyncInfo{
+		Synced:       synced,
+		PerSequencer: make(map[string]api.SequencerSyncInfo),
+	}
+	for seq := range p.Sequencers {
+		seqInfo := p.Sequencers[seq].Info()
+		ssi := api.SequencerSyncInfo{
+			Synced:           synced,
+			LatestBookedSlot: uint32(latestHealthySlot),
```
Comment (lunfardo314) [inline node/apiserver.go:63]: let's rename `LatestBookedSlot` -> `LatestHealthySlot`, `LatestSeenSlot` -> `LatestCommittedSlot` in structure and json

File: peering/peers.go [hunk 2]
```diff
@@ -344,10 +345,11 @@ func (ps *Peers) addPeer(addrInfo *peer.AddrInfo, name string, static bool) {
...
 	p := &Peer{
-		id:        addrInfo.ID,
-		name:      name,
-		isStatic:  static,
-		whenAdded: time.Now(),
+		id:            addrInfo.ID,
+		name:          name,
+		isStatic:      static,
+		whenAdded:     time.Now(),
+		peerAddresses: addrInfo.Addrs,
```
Comment (lunfardo314) [inline peering/peers.go:352]: I think you can always get peer addresses by peer ID from libp2p. Caching it here would be redundant I think

**Review by lunfardo314 — CHANGES_REQUESTED** [review]

File: node/apiserver.go (hunk 1 above)
Comment (HM4704) [reply node/apiserver.go]: Done.

File: peering/peers.go (hunk 2 above)
Comment (HM4704) [reply peering/peers.go]: Done.

**Review by lunfardo314 — APPROVED** [review]

**Pull request merged**